xbmc/guilib/TextureGL.cpp

```diff
@@ -18,6 +19,35 @@
 
 #include <memory>
 
+// Defines values that might be missing on platforms like GLES 2.0 or GL <3.3
+#ifndef GL_RED
+#define GL_RED 0x1903
+#endif
+#ifndef GL_GREEN
+#define GL_GREEN 0x1904
+#endif
+#ifndef GL_RG
+#define GL_RG 0x8227
+#endif
+#ifndef GL_TEXTURE_SWIZZLE_R
+#define GL_TEXTURE_SWIZZLE_R 0x8E42
+#endif
+#ifndef GL_TEXTURE_SWIZZLE_G
+#define GL_TEXTURE_SWIZZLE_G 0x8E43
+#endif
+#ifndef GL_TEXTURE_SWIZZLE_B
+#define GL_TEXTURE_SWIZZLE_B 0x8E44
+#endif
+#ifndef GL_TEXTURE_SWIZZLE_A
+#define GL_TEXTURE_SWIZZLE_A 0x8E45
+#endif
+#ifndef GL_TEXTURE_SWIZZLE_RGBA
+#define GL_TEXTURE_SWIZZLE_RGBA 0x8E46
+#endif
+#ifndef GL_RG_EXT
+#define GL_RG_EXT 0x8E46
```

Comment:
This value seems wrong?
```
$ grep -R -E "GL_RG(_EXT)? " /usr/include                                                                                                                                                  
/usr/include/GL/glext.h:#define GL_RG                             0x8227
/usr/include/GL/glcorearb.h:#define GL_RG                             0x8227
/usr/include/GLES2/gl2ext.h:#define GL_RG_EXT                         0x8227
/usr/include/GLES3/gl31.h:#define GL_RG                             0x8227
/usr/include/GLES3/gl3.h:#define GL_RG                             0x8227
/usr/include/GLES3/gl32.h:#define GL_RG                             0x8227
```

Reply:
Yes, it is. It was a bit late when I copied it. Also, I "GL_RG" would be more appropriate

But really, it should not matter as the code path containing it should never been taken on platforms not defining it. Should I check for that on compile time?